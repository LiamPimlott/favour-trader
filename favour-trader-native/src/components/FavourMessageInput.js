import React, { Component } from 'react';
import { StyleSheet, Text, TextInput, View } from 'react-native';
import { Button } from 'react-native-elements';

export default class FavourMessageInput extends Component {
    constructor() {
        super();
        this.state = {
            message: '',
        };
    }

    _onSubmit = () => {
        this.props.onSubmit(this.props.message);
    }

    _onCancel = () => {
        this.props.onCancel();
    }

    _onChangeText = (message) => {
        this.setState({ message });
        this.props.onChangeText(message);
    }

    render() {
        return (
            <View style={styles.columnContainer}>
                <Text>{this.props.title}</Text>
                <View style={styles.textContainer}>
                    <TextInput
                        placeholder={'Enter your message here'}
                        multiline={true}
                        numberOfLines={5}
                        onChangeText={(message) => this._onChangeText(message)}
                        value={this.props.message}
                        style={{
                            height: 400,
                            width: 310,
                            borderWidth: 1,
                            borderColor: 'black',
                            borderRadius: 2,
                            textAlignVertical: 'top',
                            marginTop: 10,
                        }}
                    />
                </View>
                <View style={styles.container}>
                    <Button
                        title={'← Offered Favours'}
                        onPress={this._onCancel}
                        buttonStyle={{
                            backgroundColor: "rgba(92, 99,216, 1)",
                            width: 125,
                            height: 45,
                            borderColor: "transparent",
                            borderWidth: 0,
                            borderRadius: 5
                        }}
                        containerStyle={{ marginTop: 20 }}
                    />
                    <Button
                        title={'Request Trade'}
                        onPress={this._onSubmit}
                        buttonStyle={{
                            backgroundColor: "rgba(92, 99,216, 1)",
                            width: 125,
                            height: 45,
                            borderColor: "transparent",
                            borderWidth: 0,
                            borderRadius: 5
                        }}
                        containerStyle={{ marginTop: 20 }}
                    />
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        paddingLeft: 10,
        paddingRight: 10,
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginTop: 10,
    },
    textContainer: {
        flexDirection: 'row',
        justifyContent: 'center',
    },
    columnContainer: {
        flexDirection: 'column',
    }
});
