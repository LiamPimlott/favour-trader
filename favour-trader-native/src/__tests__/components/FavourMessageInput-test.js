import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import moxios from "moxios";
import FavourMessageInput from '../../components/FavourMessageInput.js';
import toJson from 'enzyme-to-json'; //added this line

beforeEach(function () {
    const Adapter = require("enzyme-adapter-react-16");
    configure({adapter: new Adapter()});
    moxios.install();
});

describe('Testing FavourMessageInput component', () => {
    it('renders as expected', () => {
        const wrapper = shallow(
            <FavourMessageInput />
        );
        expect(toJson(wrapper)).toMatchSnapshot(); //edited this line
    });
});