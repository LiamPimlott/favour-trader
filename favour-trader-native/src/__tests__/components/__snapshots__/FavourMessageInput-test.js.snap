// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FavourMessageInput component renders as expected 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      numberOfLines={5}
      onChangeText={[Function]}
      placeholder="Enter your message here"
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 400,
          "marginTop": 10,
          "textAlignVertical": "top",
          "width": 310,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <Button
      TouchableComponent={[Function]}
      buttonStyle={
        Object {
          "backgroundColor": "rgba(92, 99,216, 1)",
          "borderColor": "transparent",
          "borderRadius": 5,
          "borderWidth": 0,
          "height": 45,
          "width": 125,
        }
      }
      clear={false}
      containerStyle={
        Object {
          "marginTop": 20,
        }
      }
      disabled={false}
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      title="← Offered Favours"
    />
    <Button
      TouchableComponent={[Function]}
      buttonStyle={
        Object {
          "backgroundColor": "rgba(92, 99,216, 1)",
          "borderColor": "transparent",
          "borderRadius": 5,
          "borderWidth": 0,
          "height": 45,
          "width": 125,
        }
      }
      clear={false}
      containerStyle={
        Object {
          "marginTop": 20,
        }
      }
      disabled={false}
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      title="Request Trade"
    />
  </View>
</View>
`;
